lib/oauth2.js

```diff
@@ -50,6 +56,10 @@ async function _getNewAccessToken({
   scopes,
   maxRetries = 3
 }) {
+  const cachedAccessToken = accessTokenCache.get(client_id);
```

Comment:
Suggest adding scopes to cache key, otherwise incorrect token will be used in some cases.

```suggestion
  const cachedAccessToken = accessTokenCache.get(client_id + scopes);
```

Reply:
good idea and sorry I missed that.

Reply:
Reminder to check to see if you have to canonicalize the scopes... if you don't, you might get cache misses if you have multiple scopes and different code paths that build the scopes object.